diff -ur ocamlgraph-1.7.orig/lib/bitv.ml ocamlgraph-1.7/lib/bitv.ml
--- ocamlgraph-1.7.orig/lib/bitv.ml	2022-11-18 17:43:54.907784451 +0000
+++ ocamlgraph-1.7/lib/bitv.ml	2022-11-18 17:44:01.751765389 +0000
@@ -459,11 +459,11 @@
 
 let to_string v = 
   let n = v.length in
-  let s = String.make n '0' in
+  let s = Buffer.create n in
   for i = 0 to n - 1 do
-    if unsafe_get v i then s.[i] <- '1'
+    Buffer.add_char s (if unsafe_get v i then '1' else '0')
   done;
-  s
+  Buffer.contents s
 
 let print fmt v = Format.pp_print_string fmt (to_string v)
 
Only in ocamlgraph-1.7/lib: bitv.ml.orig
